### ISSUE TITLE
Add HttpContent.SerializeToStreamAsync overload with cancellation

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#17492]
We should consider adding overloads for methods like SerializeToStreamAsync, so that cancellation can be passed in and threaded through to the underlying ReadAsync/WriteAsync operations performed.

Note: Read* methods are now covered in dotnet/runtime#916
```c#
class HttpContent {
    protected virtual Task SerializeToStreamAsync(Stream stream, TransportContext context, CancellationToken token);
    // EXISTING API:
    protected abstract Task SerializeToStreamAsync(Stream stream, TransportContext context);
}
```

**Comment by karelz** [comment]
@MihaZupan what is left to do here?

**Comment by karelz** [comment]
Fixed in #686

**Comment by MihaZupan** [comment]
There is still a comment in the code referring to this issue. I will look into it.
